.platform/hooks/predeploy/05_set_git_profile.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+# Get the env type from EB
+ENV_TYPE=$(/opt/elasticbeanstalk/bin/get-config environment -k SSM_PREFIX)
+
+GIT_USER_NAME_PARAM_NAME="${ENV_TYPE}_GIT_USER_NAME"
+GIT_USER_EMAIL_PARAM_NAME="${ENV_TYPE}_GIT_USER_EMAIL"
+
+# Set AWS region
+aws configure set default.region ap-southeast-1
+
+# Fetch git user's name and email from SSM
+GIT_USER_NAME=$(aws ssm get-parameter --name $GIT_USER_NAME_PARAM_NAME --with-decryption --query "Parameter.Value" --output text)
+GIT_USER_EMAIL=$(aws ssm get-parameter --name $GIT_USER_EMAIL_PARAM_NAME --with-decryption --query "Parameter.Value" --output text)
+
+# Write the configuration to .gitconfig
+echo "[user]" >> /home/webapp/.gitconfig
+echo "  name = $GIT_USER_NAME" >> /home/webapp/.gitconfig
+echo "  email = $GIT_USER_EMAIL" >> /home/webapp/.gitconfig
```

Comment:
Also for the file `.gitconfig` ownership and permission, if the file is created in the first run of the script, the owner might be `root` user and we need to change the ownership to `webapp` so it can access it (and set up permission).

Reply:
good point, have added logic to update to webapp user